// ngx-auth-firebaseui/esm5/lib/components/email-confirmation/email-confirmation.component.js
import { __awaiter, __decorate, __generator } from "tslib";
import { ChangeDetectionStrategy, ChangeDetectorRef, Component, EventEmitter, Input, OnChanges, OnInit, Output, SimpleChanges, TemplateRef, ViewChild } from '@angular/core';
import { Router } from '@angular/router';
import { AuthProcessService } from '../../services';
var defaultTranslations = {
    verifyEmailTitleText: 'Confirm your e-mail address!',
    verifyEmailConfirmationText: 'A confirmation e-mail has been sent.' +
        ' Check your inbox and click on the link "Confirm my e-mail" to confirm your e-mail address.',
    verifyEmailGoBackText: 'Go back',
    sendNewVerificationEmailText: 'Send new confirmation e-mail',
    signOutText: 'Sign out',
    messageOnEmailConfirmationSuccess: 'A new confirmation e-mail has been sent. Please check your inbox.',
};
var EmailConfirmationComponent = /** @class */ (function () {
    function EmailConfirmationComponent(authProcess, router, changeDetectorRef) {
        this.authProcess = authProcess;
        this.router = router;
        this.changeDetectorRef = changeDetectorRef;
        this.signOut = new EventEmitter();
    }
    EmailConfirmationComponent.prototype.ngOnChanges = function (changes) {
        if (changes.verifyEmailTemplate && changes.verifyEmailTemplate.currentValue == null) {
            this.verifyEmailTemplate = this.defaultTemplate;
            console.log('ngOnChanges - defaultTemplate:', this.verifyEmailTemplate);
        }
        this.verifyEmailContext = this.createTemplateContext();
    };
    EmailConfirmationComponent.prototype.ngOnInit = function () {
        if (!this.verifyEmailTemplate) {
            console.log('ngOnInit - defaultTemplate');
            this.verifyEmailTemplate = this.defaultTemplate;
        }
        this.verifyEmailContext = this.createTemplateContext();
        console.log('verifyEmailTemplate:', this.verifyEmailTemplate);
        console.log('verifyEmailContext:', this.verifyEmailContext);
    };
    EmailConfirmationComponent.prototype.continue = function () {
        return __awaiter(this, void 0, void 0, function () {
            var error_1;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        _a.trys.push([0, 3, , 4]);
                        return [4 /*yield*/, this.authProcess.reloadUserInfo()];
                    case 1:
                        _a.sent();
                        return [4 /*yield*/, this.router.navigate([this.goBackURL])];
                    case 2:
                        _a.sent();
                        return [3 /*break*/, 4];
                    case 3:
                        error_1 = _a.sent();
                        this.authProcess.notifyError(error_1);
                        return [3 /*break*/, 4];
                    case 4: return [2 /*return*/];
                }
            });
        });
    };
    EmailConfirmationComponent.prototype.sendNewVerificationEmail = function () {
        return __awaiter(this, void 0, void 0, function () {
            var error_2;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        _a.trys.push([0, 2, 3, 4]);
                        this.isLoading = true;
                        this.changeDetectorRef.markForCheck();
                        return [4 /*yield*/, this.authProcess.sendNewVerificationEmail()];
                    case 1:
                        _a.sent();
                        this.authProcess.showToast(this.verifyEmailContext.messageOnEmailConfirmationSuccess);
                        return [3 /*break*/, 4];
                    case 2:
                        error_2 = _a.sent();
                        this.authProcess.notifyError(error_2);
                        return [3 /*break*/, 4];
                    case 3:
                        this.isLoading = false;
                        this.changeDetectorRef.markForCheck();
                        return [7 /*endfinally*/];
                    case 4: return [2 /*return*/];
                }
            });
        });
    };
    EmailConfirmationComponent.prototype.createTemplateContext = function () {
        var context = {
            email: this.email,
            goBackURL: this.goBackURL,
            verifyEmailTitleText: this.verifyEmailTitleText || defaultTranslations.verifyEmailTitleText,
            verifyEmailConfirmationText: this.verifyEmailConfirmationText || defaultTranslations.verifyEmailConfirmationText,
            verifyEmailGoBackText: this.verifyEmailGoBackText || defaultTranslations.verifyEmailGoBackText,
            sendNewVerificationEmailText: this.sendNewVerificationEmailText || defaultTranslations.sendNewVerificationEmailText,
            signOutText: this.signOutText || defaultTranslations.signOutText,
            messageOnEmailConfirmationSuccess: this.messageOnEmailConfirmationSuccess || defaultTranslations.messageOnEmailConfirmationSuccess
        };
        return context;
    };
    EmailConfirmationComponent.ctorParameters = function () { return [
        { type: AuthProcessService },
        { type: Router },
        { type: ChangeDetectorRef }
    ]; };
    __decorate([
        Input()
    ], EmailConfirmationComponent.prototype, "email", void 0);
    __decorate([
        Input()
    ], EmailConfirmationComponent.prototype, "goBackURL", void 0);
    __decorate([
        Input()
    ], EmailConfirmationComponent.prototype, "verifyEmailTitleText", void 0);
    __decorate([
        Input()
    ], EmailConfirmationComponent.prototype, "verifyEmailConfirmationText", void 0);
    __decorate([
        Input()
    ], EmailConfirmationComponent.prototype, "verifyEmailGoBackText", void 0);
    __decorate([
        Input()
    ], EmailConfirmationComponent.prototype, "sendNewVerificationEmailText", void 0);
    __decorate([
        Input()
    ], EmailConfirmationComponent.prototype, "signOutText", void 0);
    __decorate([
        Input()
    ], EmailConfirmationComponent.prototype, "messageOnEmailConfirmationSuccess", void 0);
    __decorate([
        Input()
    ], EmailConfirmationComponent.prototype, "template", void 0);
    __decorate([
        Output()
    ], EmailConfirmationComponent.prototype, "signOut", void 0);
    __decorate([
        ViewChild('defaultVerifyEmail', { static: true })
    ], EmailConfirmationComponent.prototype, "defaultTemplate", void 0);
    EmailConfirmationComponent = __decorate([
        Component({
            selector: 'ngx-auth-firebaseui-email-confirmation',
            template: "<ng-container *ngTemplateOutlet=\"verifyEmailTemplate; context: verifyEmailContext\"></ng-container>\n<ng-template #defaultVerifyEmail let-email=\"email\" let-goBackURL=\"goBackURL\"\n             let-sendNewVerificationEmailText=\"sendNewVerificationEmailText\"\n             let-signOutText=\"signOutText\"\n             let-verifyEmailConfirmationText=\"verifyEmailConfirmationText\" let-verifyEmailGoBackText=\"verifyEmailGoBackText\"\n             let-verifyEmailTitleText=\"verifyEmailTitleText\">\n  <mat-card class=\"verify-email\">\n    <mat-card-content fxLayout=\"column\" fxLayoutAlign=\"center center\">\n      <mat-icon>email</mat-icon>\n      <p class=\"title\" fxLayout=\"column\" fxLayoutAlign=\"center center\">\n        <span class=\"mat-subheading-2\">{{ verifyEmailTitleText }}</span>\n        <span class=\"mat-body-2\">{{ email }}</span>\n      </p>\n      <p class=\"subtitle\">{{ verifyEmailConfirmationText }}</p>\n      <mat-progress-bar *ngIf=\"isLoading\" mode=\"indeterminate\"></mat-progress-bar>\n    </mat-card-content>\n    <mat-card-actions fxLayout=\"column\" fxLayoutAlign=\"center center\">\n      <button (click)=\"continue()\" *ngIf=\"goBackURL\" class=\"go-back-button action-button\" mat-stroked-button>\n        {{ verifyEmailGoBackText }}\n      </button>\n      <button (click)=\"sendNewVerificationEmail()\" class=\"send-new-mail-button action-button\"\n              mat-stroked-button>{{ sendNewVerificationEmailText }}</button>\n      <button (click)=\"this.signOut.emit()\" class=\"sign-out-button action-button\" color=\"warn\"\n              mat-stroked-button>{{ signOutText }}</button>\n    </mat-card-actions>\n  </mat-card>\n</ng-template>\n",
            changeDetection: ChangeDetectionStrategy.OnPush,
            styles: [".material-icons{font-size:4rem}.verify-email{width:360px}.verify-email .mat-icon{height:4rem;width:4rem;color:#444}.verify-email .title{margin-top:16px}.verify-email .title .mat-subheading-2{margin-bottom:0}.verify-email .subtitle{margin:16px auto;text-align:justify}.verify-email p{display:block;-webkit-margin-before:1em;-webkit-margin-after:1em;-webkit-margin-start:0;-webkit-margin-end:0}.verify-email mat-card-actions{text-align:center;margin-top:1rem}.verify-email mat-card-actions .action-button{width:100%}.verify-email mat-card-actions .action-button+.action-button{margin-top:1rem}"]
        })
    ], EmailConfirmationComponent);
    return EmailConfirmationComponent;
}());
export { EmailConfirmationComponent };
